File: pyg-sellergoods-interface/src/main/java/com/pyg/sellergoods/service/BrandService.java
package com.pyg.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pyg.entity.PageResult;
import com.pyg.entity.PygResult;
import com.pyg.pojo.TbBrand;

public interface BrandService {

	PageResult listBrand(Integer currentPage, Integer pageSize);

	//重载一个条件查询的方法
	PageResult listBrand(Integer currentPage, Integer pageSize,TbBrand brand);
	
	PygResult addBrand(TbBrand brand);

	PygResult modBrand(TbBrand brand);

	//为了编辑时的回显数据
	TbBrand getBrand(long id);
	
	//批量删除，可变参数
	PygResult delete(long ...id);
	
	//返回Map形式的品牌数据
	List<Map> listBrandOption();
}

File: pyg-manager-web/src/main/webapp/js/service/brandService.js
/**
 * 品牌管理业务层
 */
//定义业务层
	app.service("brandService", function($http) {
		//分页显示品牌列表
		this.listBrand = function(currentPage, pageSize) {
			return $http.get("../brand/listBrand.do?currentPage=" + currentPage
					+ "&pageSize=" + pageSize);
		};
		//查询单个品牌（编辑时回显数据）
		this.getBrand = function(id) {
			return $http.get("../brand/getBrand.do?id=" + id);
		};
		//添加品牌
		this.addBrand = function(entity) {
			return $http.post("../brand/addBrand.do", entity);
		};
		//修改品牌
		this.modBrand = function(entity) {
			return $http.post("../brand/modBrand.do", entity);
		};
		//删除品牌
		this.delBrand = function(ids) {
			return $http.get("../brand/delBrand.do?ids=" + ids);
		};
		//高级查询
		this.searchBrand = function(currentPage, pageSize, entity) {
			return $http.post("../brand/searchBrand.do?currentPage=" + currentPage
					+ "&pageSize=" + pageSize, entity);
		};
		//模板的下拉列表数据
		this.listBrandOption =function(){
			return $http.get("../brand/listBrandOption.do");
		}
	})